helm/platform-health/templates/role.yaml

```diff
@@ -0,0 +1,63 @@
+---
+{{- $role := .Values.role | default dict }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: platform-health
+  {{- $labels := mergeOverwrite (.Values.common_labels | default dict) ($role.labels | default dict) | toYaml | nindent 4 }}
+  labels: {{ $labels }}
+  {{- $annotations := mergeOverwrite (.Values.common_annotations | default dict) ($role.annotations | default dict) | toYaml | nindent 4 }}
+  annotations: {{ $annotations }}
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch"]
```

Comment:
Debatable whether we really need `list` and `watch`. I'd prefer to update the chart as and when we need the additional privileges, rather than requesting pre-emptively.